flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsMemoryDataManager.java

```diff
@@ -106,6 +107,30 @@ public void append(ByteBuffer record, int targetChannel, Buffer.DataType dataTyp
         }
     }
 
+    /** Close this {@link HsMemoryDataManager}, it means no data can append to memory. */
+    public void close() {
+        Decision decision = callWithLock(() -> spillStrategy.onResultPartitionClosed(this));
+        handleDecision(Optional.of(decision));
+        spiller.close();
+    }
+
+    /**
+     * Release this {@link HsMemoryDataManager}, it means all memory taken by this class will
+     * recycle.
+     */
+    public void release() {
+        spiller.release();
+    }
+
+    /**
+     * Setup components of {@link HsMemoryDataManager}.
+     *
+     * <p>This method only called by main task thread.
+     */
+    public void setup() throws IOException {
+        spiller.setup();
+    }
```

Comment:
If we are creating `HsMemoryDataManager` in `ResultPartition#setup` anyway, we should not need this method, and the relevant changes to the spiller.

Reply:
Yes, fixed.